### ISSUE TITLE
Simplistic demo how datalad can be used to track the production of derived data

### ISSUE THREAD

**Issue by mih** [datalad/datalad#1744]
```sh
datalad create demo
cd demo
# we should be able to do the next two lines via datalad
git annex addurl https://upload.wikimedia.org/wikipedia/commons/a/a5/Flower_poster_2.jpg --file sources/flowers.jpg
datalad save -m "Added flower mosaic from wikimedia"
datalad run convert -extract 1522x1522+0+0 sources/flowers.jpg flower1.jpg
datalad run convert -extract 1522x1522+1470+1470 sources/flowers.jpg flower2.jpg 
git annex whereis sources/flowers.jpg
datalad diff --revision @~1
datalad diff --revision @~3
```

**Comment by mih** [comment]
https://asciinema.org/a/134605

**Comment by mih** [comment]
Included in #1730 now
